### DISCUSSION TITLE
How to do 'px to rem' function

### DISCUSSION THREAD

**Discussion by sondh0127** [unocss/unocss#754] [Q&A]
Hey guy, currently I have a project which use twind/shim for styling. I convert some design which use px to code.

Twind provide a setup function which I can pass a function to modify the value and unit. I wonder how can I do it with uno
![image](https://user-images.githubusercontent.com/62163604/159917275-0f8ee9f5-22b6-4e9c-b9e8-0b6449a00926.png)

**Comment by antfu** [comment]
Didn't get what you mean, can you share more example?

**Reply by sondh0127** [reply]
Thank @antfu for quick response.
My goal here was to create a function similar to [`https://github.com/cuth/postcss-pxtorem`](https://github.com/cuth/postcss-pxtorem) but with atomic css
Input: with arbitrary px
![image](https://user-images.githubusercontent.com/62163604/159934920-93dc34db-3d63-4b49-ab66-f1134284e75e.png)

Output: converted to rem with root design font-size
![image](https://user-images.githubusercontent.com/62163604/159935040-1e4cebff-4c0b-44c3-bce7-f1da31ace326.png)

The way with twind:
![image](https://user-images.githubusercontent.com/62163604/159935407-10c5d03d-2054-42f5-9f1b-59f22abb5478.png)

A folk from postcss-pxtorem
![image](https://user-images.githubusercontent.com/62163604/159935567-f2e72152-6bd9-4349-bf6c-f5ff0755d0e9.png)

**Reply by antfu** [reply]
You can use this hook https://github.com/unocss/unocss/blob/b07b50d69220928eacfcf052453c2f7ce8245f47/packages/core/src/types.ts#L298-L301

**Reply by sondh0127** [reply]
I will give it a try. Thank you

**Reply by will-YT** [reply]
Hi, have you solved this problem?

**Reply by viniciusnevescosta** [reply]
Hi, has anyone actually solved this?

**Comment by antfu** [comment]
Here is a reference, but in a reversed way as you are expecting https://github.com/unocss/unocss/tree/main/packages/preset-rem-to-px

**Reply by xinlong-chen865** [reply]
@unocss/preset-rem-to-px can't be used

**Comment by 1oo1** [comment]
FYI.

```javascript
// uno.config.ts
postprocess: [
    (util) => {
      // implement postcss-pxtorem
      const pxRE = /(-?[\.\d]+)px/g;
      util.entries.forEach((entry) => {
        const value = entry[1];
        if (typeof value === 'string' && pxRE.test(value)) {
          // 100 is [rootValue] of postcss-pxtorem in my project
          // eslint-disable-next-line no-param-reassign
          entry[1] = value.replace(pxRE, (_, pixelValue) => `${pixelValue / 100}rem`);
        }
      });
    }
  ]
```

**Comment by exside** [comment]
I did it like this, it's meant being used in runtime and will automatically evaluate the pixel size of `1rem`:

```javascript
		// convert pixels to rem for all utils
		postprocess: (util, ...args) => {
			if ( !window || !window.__unocss?.theme?.px2rem ) { return }

			const px = /(-?[\.\d]+)px/g;
			const reminpx = parseFloat(getComputedStyle(document.documentElement).fontSize);
			util.entries.forEach((i) => {
				const value = i[1];
				if ( typeof value === 'string' && px.test(value) ) {
					i[1] = value.replace(px, (_, v) => `${v / reminpx}rem`);
				}
			});
		},
```

ah, I made it so that it can be enabled/disabled via a theme property `px2rem: true/false`
